include/libtorrent/torrent_info.hpp

```diff
@@ -378,8 +378,11 @@ TORRENT_VERSION_NAMESPACE_3
 		index_range<piece_index_t> piece_range() const
 		{ return m_files.piece_range(); }
 
-		// returns the info-hash of the torrent
-		const info_hash_t& info_hash() const { return m_info_hash; }
+		// returns the info-hash of the torrent. For BitTorrent v2 support, use
+		// ``info_hashes()`` to get an object that may hold both a v1 and v2
+		// info-hash
+		sha1_hash info_hash() const { return m_info_hash.get_best(); }
```

Comment:
Since the goal here is to provide backwards compatibility with unmodified clients, I think this should return the v1 hash for hybrid torrents instead of the truncated v2 hash. Non-v2 aware clients are likely to display this to the user as a v1 hash which will cause confusion when it doesn't work with v1 only clients.

Reply:
makes sense. I was thinking that, as long as the info-hash is just used as a unique identifier, a client could support v2 torrents without knowing about it. but maybe that's risky